int16[6] pulses
